src/extlibs/hrvo/simulator.cpp

```diff
@@ -64,73 +62,43 @@ void HRVOSimulator::updateWorld(const World &world)
     this->world               = world;
     const auto &friendly_team = world.friendlyTeam().getAllRobots();
     const auto &enemy_team    = world.enemyTeam().getAllRobots();
-    // TODO (#2498): Update implementation to correctly support adding and removing agents
-    //               to represent the newly added and removed friendly/enemy robots in the
-    //               World.
-    if (friendly_robot_id_map.empty() && enemy_robot_id_map.empty())
+
+    updateRemovedAgents(world);
```

Comment:
We want a simpler implementation for `updateWorld`, without `updateRemovedAgents` as mentioned in a comment further above.